x-pack/test/detection_engine_api_integration/security_and_spaces/tests/keyword_family/keyword_mixed_with_const.ts

```diff
@@ -0,0 +1,171 @@
+      await esArchiver.load('rule_keyword_family/keyword');
+    });
+
+    afterEach(async () => {
+      await deleteSignalsIndex(supertest);
+      await deleteAllAlerts(supertest);
+      await esArchiver.unload('rule_keyword_family/const_keyword');
+      await esArchiver.unload('rule_keyword_family/keyword');
+    });
+
+    describe('"kql" rule type', () => {
+      it('should detect the "dataset_name_1" from "event.dataset" and have 8 signals, 4 from each index', async () => {
+        const rule = {
+          ...getRuleForSignalTesting(['keyword', 'const_keyword']),
+          query: 'event.dataset: "dataset_name_1"',
+        };
+        const { id } = await createRule(supertest, rule);
+        await waitForRuleSuccessOrStatus(supertest, id);
+        await waitForSignalsToBePresent(supertest, 8, [id]);
+        const signalsOpen = await getSignalsById(supertest, id);
+        expect(signalsOpen.hits.hits.length).to.eql(8);
+      });
+
+      // TODO: Fix this bug and make this work. We currently do not write out the dataset name when it is not in _source
+      it.skip('should copy the dataset_name_1 from the index into the signal', async () => {
+        const rule = {
+          ...getRuleForSignalTesting(['keyword', 'const_keyword']),
+          query: 'event.dataset: "dataset_name_1"',
+        };
+        const { id } = await createRule(supertest, rule);
+        await waitForRuleSuccessOrStatus(supertest, id);
+        await waitForSignalsToBePresent(supertest, 8, [id]);
+        const signalsOpen = await getSignalsById(supertest, id);
+        const hits = signalsOpen.hits.hits
+          .map((hit) => (hit._source.event as EventModule).dataset)
+          .sort();
+        expect(hits).to.eql([
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+          'dataset_name_1',
+        ]);
+      });
+    });
+
+    describe('"eql" rule type', () => {
+      it('should detect the "dataset_name_1" from "event.dataset" and have 8 signals, 4 from each index', async () => {
+        const rule: EqlCreateSchema = {
+          ...getRuleForSignalTesting(['keyword', 'const_keyword']),
+          rule_id: 'eql-rule',
+          type: 'eql',
+          language: 'eql',
+          query: 'any where event.dataset=="dataset_name_1"',
+        };
+
+        const { id } = await createRule(supertest, rule);
+        await waitForRuleSuccessOrStatus(supertest, id);
+        await waitForSignalsToBePresent(supertest, 8, [id]);
+        const signalsOpen = await getSignalsById(supertest, id);
+        expect(signalsOpen.hits.hits.length).to.eql(8);
+      });
+
+      // TODO: Fix this bug and make this work. We currently do not write out the dataset name when it is not in _source
+      it.skip('should detect the "dataset_name_1" from "event.dataset"', async () => {
```

Comment:
Same comment as above. Should the test description here be something like `should copy the dataset_name_1 from the index into the signal` instead?

Reply:
Yeah, that would make more sense. I will change it.